packages/toolkit/core-kit/src/password-policy.ts

```diff
@@ -282,171 +313,143 @@ export class PasswordPolicyChecker {
   }
 
   /**
-   * Check if the given password contains repetition characters that are more than
-   * the threshold.
+   * Get the length of the repetition at the beginning of the given string.
+   * For example, `repetitionLength('aaaaa')` will return `5`.
    *
-   * @param password - Password to check.
-   * @returns Whether the password contains repetition characters.
+   * If the length is less than {@link PasswordPolicyChecker.repetitionAndSequenceThreshold},
+   * `0` will be returned.
    */
-  hasRepetition(password: string): boolean {
-    const { repetitionAndSequenceThreshold, getRestrictedPhraseThreshold } = PasswordPolicyChecker;
+  /* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
+  repetitionLength(password: string): number {
+    const { repetitionAndSequenceThreshold } = PasswordPolicyChecker;
+    const firstChar = password[0]?.toLowerCase();
+    let length = 0;
 
-    if (password.length < repetitionAndSequenceThreshold) {
-      return false;
+    for (const char of password) {
+      if (char.toLowerCase() === firstChar) {
```

Comment:
A combination of the lower case and upper case is considered as repetition?

Reply:
updated